baybe/targets/base.py

```diff
@@ -38,17 +38,14 @@ def to_objective(self) -> SingleTargetObjective:
         return SingleTargetObjective(self)
 
     @abstractmethod
-    def transform(self, data: pd.DataFrame) -> pd.DataFrame:
-        """Transform data into computational representation.
-
-        The transformation depends on the target mode, e.g. minimization, maximization,
-        matching, etc.
+    def transform(self, series: pd.Series, /) -> pd.Series:
```

Comment:
Can you explain why this is being changed from dataframe to series? Is it because we implicitly already assumed that the data was one-dimensional and hence de facto already a series or is there another reason I am not aware of?

Reply:
Yes, the assumption was already baked into the code and we always took just processed the first column, having to do an avoidable check that the input really contains only one column. I currently don't see why we would ever need it to be a dataframe. The only hypothetical scenario (which I doubt would ever come) is if we introduced some form of multidimensional targets. But even if we wanted to have something like that, we'd probably rather represent the multidimensionality into several individual targets in the outer scope or apply another mechanism. If really needed at some point, we can always generalize stuff. But for the current code version, it makes no sense at all, if you ask me. @Scienfitz: agree?

Reply:
dont think this change to series is very important, but logically it makes sense